### DISCUSSION TITLE
How to minimize excessive white space around the pyplot

### DISCUSSION THREAD

**Discussion by nghia-vo** [zauberzeug/nicegui#1773] [Q&A] [answered]
### Question

Example:
```python
import numpy as np
import matplotlib.pyplot as plt
from nicegui import ui

def show_data():
    with main_plot:
        if plot_list.value == "imshow":
            with main_plot:
                data = np.random.rand(120, 100)
                plt.clf()
                plt.imshow(data)
        else:
            with main_plot:
                data = np.random.rand(100)
                plt.clf()
                plt.plot(data)
with ui.row():
    ui.button('Show data', on_click=lambda: show_data())
    plot_list = ui.select(["imshow", "plot"], value="imshow")
ui.separator()
main_plot = ui.pyplot(figsize=[10, 8], close=False)
ui.separator()

ui.run()
```
How to reduce white spaces as below?
![image](https://github.com/zauberzeug/nicegui/assets/12136997/350236b1-d6b5-4abb-8c17-c90da53d37fc)
![image](https://github.com/zauberzeug/nicegui/assets/12136997/2ae28fcb-7120-4ba7-9c37-121d38b7ecb7)
Many thanks.

**Comment by falkoschindler** [comment, marked as answer]
You can add [`plt.tight_layout()`](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.tight_layout.html) after adding the image or the plot to the figure.

**Reply by nghia-vo** [reply]
Ha, I used that but before plt.imshow()/plt.plot() and saw no effect. Thanks for the tip.
